File: frontend/src/types/drone.ts
/** TypeScript interfaces for Drone entity matching Pydantic schemas. */

import type { DroneStatus } from './enums';

export interface DroneResponse {
  id: number;
  identificacao: string;
  modelo: string;
  fabricante: string | null;
  capacidade_litros: number;
  status: DroneStatus;
  horas_voadas: number;
  ultima_manutencao_em: string | null;
  ativo: boolean;
  created_at: string;
  updated_at: string;
}

export interface DroneCreate {
  identificacao: string;
  modelo: string;
  fabricante?: string | null;
  capacidade_litros: number;
  status?: DroneStatus;
}

export interface DroneUpdate {
  identificacao?: string | null;
  modelo?: string | null;
  fabricante?: string | null;
  capacidade_litros?: number | null;
  status?: DroneStatus | null;
}

File: app/schemas/drone.py
"""Pydantic schemas for Drone entity."""

from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, Field

from app.models.enums import DroneStatus


class DroneBase(BaseModel):
    identificacao: str = Field(..., max_length=120)
    modelo: str = Field(..., max_length=120)
    fabricante: str | None = Field(None, max_length=120)
    capacidade_litros: Decimal = Field(..., ge=0)
    status: DroneStatus = DroneStatus.DISPONIVEL


class DroneCreate(DroneBase):
    pass


class DroneUpdate(BaseModel):
    identificacao: str | None = Field(None, max_length=120)
    modelo: str | None = Field(None, max_length=120)
    fabricante: str | None = Field(None, max_length=120)
    capacidade_litros: Decimal | None = Field(None, ge=0)
    status: DroneStatus | None = None


class DroneResponse(DroneBase):
    id: int
    horas_voadas: Decimal
    ultima_manutencao_em: date | None
    ativo: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
